core/data_encoder_decoder.py

```diff
@@ -55,9 +55,6 @@ def encode(self, data_stream: DataStream, block_size: int, encode_writer: Encode
             encode_writer (EncodedBlockWriter): the writer used to write encoded bitarrays
         """
 
-        # reset the state
-        self.reset()
```

Comment:
removed reset here and in decoder because that deleted the initialized window in LZ77. I think doesn't make sense to reset right after creation anyway. Please correct me if I am missing anything.